tests/testthat/test-test-threads.R

```diff
@@ -0,0 +1,222 @@
+
+    search_str <- "MAXEVAL|NITER"
+    max_evals <- map(mods, function(.mod){
+      mod_path <- get_model_path(.mod)
+      mod_lines <- mod_path %>% readLines()
+      str_line_loc <- which(grepl(search_str, mod_lines))
+      str_values <- str_split(mod_lines[str_line_loc], " ")[[1]]
+      str_loc <- grepl(search_str, str_values)
+      as.numeric(gsub('MAXEVAL=', '', str_values[str_loc]))
+    }) %>% unlist()
+
+    expect_true(all(max_evals == 100))
+  })
+
+
+  test_that("test_threads(.dry_run=T) threads are set correctly [BBR-TSTT-003]", {
+    mod_threads <- lapply(mods, function(mod.x){mod.x$bbi_args$threads}) %>% unlist()
+    expect_equal(mod_threads[1], 2)
+    expect_equal(mod_threads[2], 4)
+  })
+
+
+  mod_none <- copy_model_from(
+    read_model(file.path(MODEL_DIR_BBI, "1")),
+    "none"
+  ) %>% add_tags("no maxeval")
+
+  mod_both <- copy_model_from(
+    read_model(file.path(MODEL_DIR_BBI, "1")),
+    "both"
+  ) %>% add_tags("maxeval and niter")
+
+  mods_fake <- list(mod_none, mod_both)
+
+
+  test_that("test_threads(.dry_run=T) correctly errors out if no maxeval, or both maxeval and niter are provided [BBR-TSTT-004]", {
+
+    search_str <- "MAXEVAL|NITER"
+    str_replacements <- c("", "MAXEVAL=10 NITER=10")
+
+    map2(mods_fake, str_replacements, function(.mod, str_replace){
+      mod_path <- get_model_path(.mod)
+      mod_lines <- mod_path %>% readLines()
+      str_line_loc <- which(grepl(search_str, mod_lines))
+      str_values <- str_split(mod_lines[str_line_loc], " ")[[1]]
+      str_loc <- grepl(search_str, str_values)
+      str_line_update <- paste(paste(str_values[!str_loc], collapse = " "), str_replace, sep = " ")
+      mod_lines[str_line_loc] <- str_line_update
+      writeLines(mod_lines, mod_path)
+    })
+
+    expect_error(
+      test_threads(mod_none, .threads = c(2, 4), .max_eval = 100, .mode = "local", .dry_run = TRUE),
+      "Neither MAXEVAL or NITER were found in the ctl file. Please ensure one is provided")
+    expect_error(
+      test_threads(mod_both, .threads = c(2, 4), .max_eval = 100, .mode = "local", .dry_run = TRUE),
+      "Both MAXEVAL and NITER were found in the ctl file. Please ensure only one is provided")
+  })
+
+
+  test_that("check_run_times() returns NA for dry runs [BBR-TSTT-005]", {
+    expect_equal(check_run_times(mod1, .wait = F), NA)
+  })
+
+  test_that("cleanup_mods() works for models created by test_threads by default [BBR-CLM-001]", {
+
+    mod_ctls <- lapply(mods, function(mod.x){get_model_path(mod.x)}) %>% unlist()
+    mod_yamls <- lapply(mods, function(mod.x){mod.x$absolute_model_path}) %>% yaml_ext()
+
+    msg_remove <- paste0(
+      paste("Removed", length(mods), "models with the following tags:\n"),
+      paste("-","test threads", collapse = "\n")
+    )
+    expect_message(
+      cleanup_mods(mods, .force = T),
+      msg_remove
+    )
+
+    for(i in 1:length(mod_ctls)){
```

Comment:
`file_exists()` is vectorized, so you can just do

```r
expect_false(any(fs::file_exists(mod_ctls)))
expect_false(any(fs::file_exists(mod_yamls)))
```

and not do the `for` loop at all.

Same for a few places further down in this file.